sdk/cognitivelanguage/Azure.AI.Language.QuestionAnswering/MigrationGuide.md

```diff
@@ -0,0 +1,111 @@
+
+## General changes
+
+### Package and namespaces
+
+Package names and the namespace root for the modern Azure client libraries for .NET have changed. Each will follow the pattern `Azure.[Area].[Services]` where the legacy clients followed the pattern `Microsoft.Azure.[Service]`. This provides a quick and accessible means to help understand, at a glance, whether you are using the modern or legacy clients.
+
+In the case of Question Answering, the modern client libraries have packages and namespaces that begin with `Azure.AI.Language.QuestionAnswering` and were released beginning with version 1. The legacy client libraries have packages and namespaces that begin with `Microsoft.Azure.CognitiveServices.Knowledge.QnAMaker` and a version of 2.x.x or below.
+
+### Authenticating
+
+Previously in `Microsoft.Azure.CognitiveServices.Knowledge.QnAMaker`, you could create a `QnAMakerRuntimeClient` along with `EndpointKeyServiceClientCredentials` and `HttpClient`:
+
+```C# Snippet:CognitiveServices_QnA_Maker_Snippets_MigrationGuide_CreateRuntimeClient
+var client = new QnAMakerRuntimeClient(new EndpointKeyServiceClientCredentials("<QnAMakerEndpointKey>"))
+{
+    RuntimeEndpoint = "https://sk4cs.azurewebsites.net"
+};
+```
+
+Now in `Azure.AI.Language.QuestionAnswering`, you create a `QuestionAnsweringClient` along with `AzureKeyCredential` from the package `Azure.Core`:
+
+```C# Snippet:QuestionAnsweringClient_Create
+Uri endpoint = new Uri("https://myaccount.api.cognitive.microsoft.com");
+AzureKeyCredential credential = new AzureKeyCredential("{api-key}");
+
+QuestionAnsweringClient client = new QuestionAnsweringClient(endpoint, credential);
+```
+
+### Querying a question
+
+Previously in `Microsoft.Azure.CognitiveServices.Knowledge.QnAMaker`, you could query for a question using `QueryDTO`:
+
+```C# Snippet:CognitiveServices_QnA_Maker_Snippets_MigrationGuide_QueryKnowledgeBase
+QueryDTO queryDTO = new QueryDTO();
+queryDTO.Question = "hello";
+var answer = await client.Runtime.GenerateAnswerAsync("<KnowledgeBaseID>", queryDTO);
+```
+
+Now in `Azure.AI.Language.QuestionAnswering` you use `QueryKnowledgeBaseOptions` to set the options then you query using the client:
+
+```C# Snippet:QuestionAnsweringClient_QueryKnowledgeBaseAsync
+string projectName = "FAQ";
+string deploymentName = "prod";
+QueryKnowledgeBaseOptions options = new QueryKnowledgeBaseOptions(projectName, deploymentName, "How long should my Surface battery last?");
+Response<KnowledgeBaseAnswers> response = await client.QueryKnowledgeBaseAsync(options);
```

Comment:
This looks more complicated. I'd use the overload that takes 3 strings since that's all you need anyway, and pass the strings inline instead of separate variables like you do for the track 1 code above with the placeholder but, again, use a consistent placeholder or fake value across all samples.